app/components/BuyNBTC/BuyNBTC.tsx

```diff
@@ -6,38 +6,87 @@ import { WalletContext } from "~/providers/ByieldWalletProvider";
 import { ByieldWallet } from "~/types";
 import { SuiModal } from "../Wallet/SuiWallet/SuiModal";
 import { useCurrentAccount, useSignAndExecuteTransaction, useSuiClient } from "@mysten/dapp-kit";
-import { Transaction } from "@mysten/sui/transactions";
 import { useNetworkVariables } from "~/networkConfig";
 import { createBuyNBTCTxn } from "~/util/util";
-import { BUFFER_BALANCE, PRICE_PER_NBTC_IN_SUI } from "~/constant";
+import { BUY_NBTC_GAS_FEE_IN_SUI, PRICE_PER_NBTC_IN_SUI } from "~/constant";
 import { ArrowDown } from "lucide-react";
-import { useSuiBalance } from "../Wallet/SuiWallet/useSuiBalance";
-import { FormNumericInput } from "../form/FormNumericInput";
-import { Modal } from "../ui/dialog";
 import { formatSUI, parseSUI } from "~/lib/denoms";
 import { useToast } from "~/hooks/use-toast";
 import { useNBTCBalance } from "../Wallet/SuiWallet/useNBTCBalance";
 import { NBTCBalance } from "./NBTCBalance";
+import { FormNumericInput } from "../form/FormNumericInput";
+import { Modal } from "../ui/dialog";
+import { useSuiBalance } from "../Wallet/SuiWallet/useSuiBalance";
 import { Instructions } from "./Instructions";
-import { SUIGasFee } from "./SUIGasFee";
 import { TransactionStatus } from "./TransactionStatus";
 
+const calculateYouReceive = (_mistAmount: bigint, _gasFee: bigint): number => {
```

Comment:
This is not how I explained in the chat. `_gasFee` charged from the wallet. You should provide the `mistAmount` only. Also, remove `_` from the name.